src/libdecaf/src/modules/gx2/gx2_shaders.cpp

```diff
@@ -579,6 +601,54 @@ GX2SetStreamOutEnable(BOOL enable)
 }
 
 void
+GX2SetStreamOutContext(uint32_t index,
+                       GX2OutputStream *stream,
+                       GX2StreamOutContextMode mode)
+{
+   decaf_check(index <= 3);
+
+   auto control = pm4::SBU_CONTROL::get(0)
+      .STORE_BUFFER_FILLED_SIZE().set(false)
+      .SELECT_BUFFER().set(index);
+   auto srcLo = 0u;
+
+   switch (mode) {
+   case GX2StreamOutContextMode::Append:
+      control = control
+         .OFFSET_SOURCE().set(pm4::STRMOUT_OFFSET_FROM_MEM);
+      srcLo = stream->context.getAddress();
+      break;
+   case GX2StreamOutContextMode::FromStart:
+      control = control
+         .OFFSET_SOURCE().set(pm4::STRMOUT_OFFSET_FROM_PACKET);
+      srcLo = 0u;
+      break;
+   case GX2StreamOutContextMode::FromOffset:
+      control = control
+         .OFFSET_SOURCE().set(pm4::STRMOUT_OFFSET_FROM_PACKET);
+      srcLo = 0u;  // TODO: where does the offset come from?
```

Comment:
Hmm, if that's the case then the offset mode needs to be something else, because FROM_PACKET would try to use the memory address as an offset into the buffer (which is of course going to overflow). Can you doublecheck, or maybe put up the disassembly somewhere so I can take a look at it?

It could of course be a bug in GX2 itself...